assignments/array-methods.js

```diff
@@ -56,28 +58,66 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
-console.log(fullName);
+
+runners.forEach(function(eaRunner) {
+    fullName.push(`${eaRunner.first_name} ${eaRunner.last_name}`);
+})
+
+console.log("challenge 1", fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
-console.log(allCaps); 
+
+runners.map(function(eaRunner) {
+    allCaps.push(`${eaRunner.first_name}`.toUpperCase());
+})
+
+console.log("challenge 2", allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
-console.log(largeShirts);
+
+const lRunner = runners.filter( runner => runner.shirt_size === "L");
+largeShirts.push(lRunner);
+
+console.log("challenge 3", largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
 let ticketPriceTotal = [];
-console.log(ticketPriceTotal);
+
+const total = runners.reduce(function(accumulator, currentValue) {
+    return accumulator + currentValue.donation;
+}, 0);
+
+ticketPriceTotal.push(total);
+
+console.log("challenge 4", ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+//sort runners by donation amount without changing original array
+let runnersCopy = runners.slice().sort((a, b) => a.donation - b.donation)
+
+console.log("challenge 5.1", runnersCopy)
 
```

Comment:
immutability! oh man you don't give yourself enough credit in standup, brandi - this is fantastic and prepares you so so well for future parts of the curriculum (especially redux for react state management)!